arwen/contexts/async_synchronous.go

```diff
@@ -0,0 +1,243 @@
+package contexts
+
+import (
+	"math/big"
+
+	"github.com/ElrondNetwork/arwen-wasm-vm/arwen"
+	"github.com/ElrondNetwork/elrond-go/core/vmcommon"
+)
+
+func (context *asyncContext) executeSynchronousCalls() error {
+	for groupIndex, group := range context.AsyncCallGroups {
+		for _, call := range group.AsyncCalls {
+			if call.ExecutionMode != arwen.SyncExecution {
+				continue
+			}
+
+			err := context.executeSyncCall(call)
+			if err != nil {
+				return err
+			}
+		}
+
+		group.DeleteCompletedAsyncCalls()
+
+		// If all the AsyncCalls in the AsyncCallGroup were executed synchronously,
+		// then the AsyncCallGroup can have its callback executed.
+		if group.IsCompleted() {
+			context.executeCallGroupCallback(group)
+			context.DeleteCallGroup(groupIndex)
+		}
+	}
+	return nil
+}
+
+func (context *asyncContext) executeSyncCall(asyncCall *arwen.AsyncCall) error {
+	destinationCallInput, err := context.createSyncCallInput(asyncCall)
+	if err != nil {
+		return err
+	}
+
+	vmOutput, err := context.host.ExecuteOnDestContext(destinationCallInput)
```

Comment:
If err!=nil check is not needed, use "_" instead of "err".

Reply:
`err` is passed as argument to `context.executeSyncCallback()` below.